js/ui/action_button/action_button_base.js

```diff
@@ -0,0 +1,223 @@
+        const lastActionIndex = actions.length - 1;
+
+        if(actions.length >= this.option("maxActionButtonCount")) return;
+
+        if(this._actionItems) {
+            this._actionItems.forEach(actionItem => {
+                actionItem.dispose();
+                actionItem.element().remove();
+            });
+        }
+
+        if(!actions.length) return;
+
+        for(let i = 0; i < actions.length; i++) {
+            const action = actions[i];
+            const $actionElement = $("<div>")
+                .appendTo(swatch.getSwatchContainer(action.$element()));
+
+            eventsEngine.off($actionElement, "click");
+            eventsEngine.on($actionElement, "click", () => {
+                this._clickHandler();
+            });
+
+            const actionOffsetY = this.option("indent") + this.option("childIndent") * i;
+
+            action._options.position = {
+                of: this.content(),
+                at: "center",
+                my: "center",
+                offset: {
+                    x: 0,
+                    y: -actionOffsetY
+                }
+            };
+
+            action._options.animation.show.delay = action._options.animation.show.duration * i;
+            action._options.animation.hide.delay = action._options.animation.hide.duration * (lastActionIndex - i);
+
+            this._actionItems.push(this._createComponent($actionElement, ActionButtonItem, action._options));
+        }
+    },
+
+    _optionChanged(args) {
+        switch(args.name) {
+            case "actions":
+                this._renderActions();
+                break;
+            case "position":
+                this._renderActions();
+                break;
+            case "maxActionButtonCount":
+                this._renderActions();
+                break;
+            case "icon":
+                this._renderIcon();
+                break;
+            case "closeIcon":
+                this._renderCloseIcon();
+                break;
+            default:
+                this.callBase(args);
+        }
+    }
+});
+
+exports.init = function(newAction) {
+    let isActionExist = false;
+    if(!actionButtonBase) {
+        const $fabMainElement = $("<div>")
+            .appendTo(swatch.getSwatchContainer(newAction.$element()));
+        actionButtonBase = new ActionButtonBase($fabMainElement, {
+            icon: newAction._options.icon || config().actionButtonConfig.icon,
+            onClick: newAction._options.onClick,
+            actions: [ newAction ]
+        });
+    } else {
+        const savedActions = actionButtonBase.option("actions");
+
+        savedActions.forEach(action => {
+            if(action._options.id === newAction._options.id) {
+                isActionExist = true;
+                return newAction;
+            }
+        });
+
+        if(!isActionExist) {
+            savedActions.push(newAction);
+            actionButtonBase.option("actions", savedActions);
```

Comment:
Can we use [this signature](https://js.devexpress.com/Documentation/ApiReference/UI_Widgets/dxAccordion/Methods/#optionoptions) to update all options in one time?